subsys/sensing/sensor_mgmt.h

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) 2023 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef SENSOR_MGMT_H_
+#define SENSOR_MGMT_H_
+
+#include <zephyr/sensing/sensing_datatypes.h>
+#include <zephyr/sensing/sensing_sensor.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/slist.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define PHANDLE_DEVICE_BY_IDX(idx, node, prop)				\
+	DEVICE_DT_GET(DT_PHANDLE_BY_IDX(node, prop, idx))
+
+#define PHANDLE_DEVICE_LIST(node, prop)					\
+{									\
+	LISTIFY(DT_PROP_LEN_OR(node, prop, 0),				\
+			PHANDLE_DEVICE_BY_IDX,				\
+			(,),						\
+			node,						\
+			prop)						\
+}
+
+#define SENSING_SENSOR_DT_INFO(node)							\
+	static struct sensing_dt_info							\
+		_CONCAT(__sensing_dt_info_, Z_DEVICE_DT_DEV_ID(node)) = {		\
+			.dev = DEVICE_DT_GET(node),					\
+			.ord = DT_DEP_ORD(node),					\
+			.info.type = DT_PROP(node, sensor_type),			\
+			.info.name = DT_NODE_FULL_NAME(node),				\
+			.info.friendly_name = DT_PROP(node, friendly_name),		\
+			.info.minimal_interval = DT_PROP(node, minimal_interval),	\
+			.reporter_num = DT_PROP_LEN_OR(node, reporters, 0),		\
+			.reporters = PHANDLE_DEVICE_LIST(node, reporters),		\
+		};
```

Comment:
See below, I confirmed this change works with the changes to sensor_mgmt.c

```suggestion
#define SENSING_SENSOR_DT_INFO(node)							\
		{		\
			.dev = DEVICE_DT_GET(node),					\
			.ord = DT_DEP_ORD(node),					\
			.info.type = DT_PROP(node, sensor_type),			\
			.info.name = DT_NODE_FULL_NAME(node),				\
			.info.friendly_name = DT_PROP(node, friendly_name),		\
			.info.minimal_interval = DT_PROP(node, minimal_interval),	\
			.reporter_num = DT_PROP_LEN_OR(node, reporters, 0),		\
			.reporters = PHANDLE_DEVICE_LIST(node, reporters),		\
		},
```

Reply:
[0001-add-hinge-angle-virtual-sensor.patch](https://github.com/zephyrproject-rtos/zephyr/files/11575990/0001-add-hinge-angle-virtual-sensor.patch)